tests/cypress/e2e/wp/form.cy.js

```diff
@@ -54,4 +54,112 @@ describe("MFP Work Plan E2E Submission", () => {
 
     cy.contains("Successfully Submitted").should("be.visible");
   });
+
+  it("Admin users can deny a Work Plan submission", () => {
+    cy.authenticate("adminUser");
+    cy.navigateToHomePage();
+    cy.get(
+      '[aria-label="List of states, including District of Columbia and Puerto Rico"]'
+    ).select("DC");
+    cy.get('[id="report-WP"]').click();
+    cy.contains("Go to Report Dashboard").click();
+    cy.wait(3000);
+
+    //There is a new submitted Workplan with the title
+    cy.contains(
+      `District of Columbia MFP Work Plan ${currentYear} - Period 1`,
+      { matchCase: true }
+    ).should("be.visible");
+
+    //Find our new program and open it
+    cy.get("table").within(() => {
+      cy.get("td")
+        .as("workplan")
+        .contains(
+          `District of Columbia MFP Work Plan ${currentYear} - Period 1`
+        );
+      cy.get("@workplan").parent().as("workplancontainer");
+      cy.get("@workplancontainer")
+        .find('button:contains("View")')
+        .as("viewButton");
+      cy.get("@viewButton").first().focus();
+      cy.get("@viewButton").first().click();
+    });
+
+    cy.get(".chakra-link .level-1")
+      .filter(":visible")
+      .eq(3)
+      .parent()
+      .parent()
+      .click();
+    cy.get(".chakra-link .level-1")
+      .filter(":visible")
+      .eq(3)
+      .parent()
+      .should("have.class", "selected");
+
+    cy.contains("Review & Submit").should("be.visible");
+
+    cy.get(`button:contains("Unlock")`).focus().click();
+    cy.wait(1500);
+
+    cy.get('button:contains("Return to dashboard")').focus().click();
+
+    cy.contains("In revision").should("be.visible");
+  });
+
+  it("State users can update and submit an in revision Work Plan", () => {
+    cy.authenticate("stateUser");
+
+    //Given I've logged in
+    cy.url().should("include", "/");
+
+    //And I enter the Work Plan Dashboard
+    cy.contains("Enter Work Plan online").click();
+    cy.url().should("include", "/wp");
+
+    //Find our new program and open it
+    cy.get("table").within(() => {
+      cy.get("td")
+        .as("workplan")
+        .contains(
+          `District of Columbia MFP Work Plan ${currentYear} - Period 1`
+        );
+      cy.get("@workplan").parent().as("workplancontainer");
+      cy.get("@workplancontainer")
+        .find('button:contains("Edit")')
+        .as("editButton");
+      cy.get("@editButton").first().focus();
+      cy.get("@editButton").first().click();
+    });
+
+    //Using the json as a guide, traverse all the routes/forms and fill it out dynamically
+    cy.get(".chakra-link .level-1")
+      .filter(":visible")
+      .eq(3)
+      .parent()
+      .parent()
+      .click();
+    cy.get(".chakra-link .level-1")
+      .filter(":visible")
+      .eq(3)
+      .parent()
+      .should("have.class", "selected");
+
+    cy.contains("Review & Submit").should("be.visible");
```

Comment:
Should we check the URL here instead? I am wondering if this evaluates to true even if the sidebar contains this text.

Reply:
Updated to check the URL